### PR TITLE
Share component code cleanup and refactor

### PR DESCRIPTION
**Pull request by mikemai2awesome**
## Jira

http://vjira2:8080/browse/BDS-311

## Summary

Refactored share component's code to be more efficient. `inline` and `copyToClipboard` have been deprecated. `size` and `opacity` have been added as props to allow more flexibility. 

## Details

1. Cleaned up the CSS and twig file.
2. Removed as much overrides as possible.
3. Followed the Bolt conventions and renamed certain props and variables.
4. Button version has been removed because it's not being used.
5. Latest 2.x has been pulled in.
6. Icon component has been changed: small icon size has been increased and stroke has been decreased.
7. Copy to clipboard has been changed: misc. code clean ups.
8. `copyToClipboard` prop has been renamed to `copy_to_clipboard`.
9. `inline` prop has been removed because button version is not being utilized.
10. `size` prop has been added to control text and icon size, and spacing.
11. `opacity` prop has been added to control the initial transparency of the share tool.

[View demo](https://feature-bds-311-share-component-refactor.bolt-design-system.com/pattern-lab/?p=viewall-components-share)

screen shot 2018-09-25 at 12 01 38 pm

## How to test

Pull down the branch and run locally. Navigate to pattern lab and view the share component. Test all the links and copy to clipboard function.

## Notes

1. It has not been converted to a web component.

### CODE REVIEW

File: packages/components/bolt-share/src/share.twig [hunk 1]
```diff
@@ -1,87 +1,100 @@
...
+        target: source_target
+      }
+    })]) %}
   {% endif %}
 {% endfor %}
 
-{% if copyToClipboard %}
-  {% set item %}
-    {% include "@bolt-components-copy-to-clipboard/copy-to-clipboard.twig" with copyToClipboard only %}
+{% if copy_to_clipboard %}
```
Comment (sghoweri) [inline packages/components/bolt-share/src/share.twig]: @mikemai2awesome why don't you just pass in the `copy_to_clipboard` data directly to that component? 

That should eliminate the need to map anything (and would also mean the copy_to_clipboard component could get updates that wouldn't require going back in and updating the share component).

Have you tried something like this?
```twig
{% if copy_to_clipboard %}
  {% set copy_to_clipboard_include %}
    <div class="c-bolt-share__copy">
      {% include "@bolt-components-copy-to-clipboard/copy-to-clipboard.twig" with copy_to_clipboard only %}
    </div>
  {% endset %}
  {% set items = items|merge([item]) %}
  {% set items = items|merge([copy_to_clipboard_include]) %}
{% endif %}
```

That should allow you to then do something like this when you're trying to use it (note the "text" prop is nested inside of the copy_to_clipboard data):

```twig
{% include "@bolt-components-share/share.twig" with {
  sources: [
    {
      name: "facebook",
      url: "https://www.facebook.com/sharer/sharer.php?u=https://5abd3062df99537b79e29496--bolt-design-system.netlify.com/pattern-lab/patterns/02-components-action-blocks-05-action-blocks/02-components-action-blocks-05-action-blocks.html&amp;src=sdkpreparse"
    },
    {
      name: "twitter",
      url: "https://twitter.com/intent/tweet?url=https://bolt-design-system.com&text=Sample%20Share%20Text&via=pega&hashtags=boltDesignSystemRocks!"
    },
    {
      name: "linkedin",
      url: "https://www.linkedin.com/shareArticle?url=https://5abd3062df99537b79e29496--bolt-design-system.netlify.com/pattern-lab/patterns/02-components-action-blocks-05-action-blocks/02-components-action-blocks-05-action-blocks.html"
    },
    {
      name: "email",
      url: "mailto:?&body=Sample%20Text%20--%20https%3A//bolt-design-system.com"
    }
  ],
  copy_to_clipboard: {
    url: "http://pega.com",
    text: "Copy share link"
  }
} only %}
```

File: packages/components/bolt-share/src/share.twig [hunk 2]
```diff
@@ -1,87 +1,100 @@
...
-{% set attributes = create_attribute(attributes|default({})) %}
-{% set classes = [
-  "c-bolt-share",
-  visibility is sameas(false) ? "c-bolt-share--hidden",
-  isFlat is sameas(false) ? "c-bolt-share--normal-layout" : "c-bolt-share--inline-layout"
-] %}
+{# Share component's custom element wrapper. #}
+<bolt-share
+  {% if display %} display="{{ display }}" {% endif %}
+  {{ parentAttributes }}
```
[the remark this answers is not in the document]
Comment (mikemai2awesome) [reply packages/components/bolt-share/src/share.twig]: Things like hidden/visuallyhidden, margin, and padding overrides need to be on the outer most container for most scenarios to work as expected.

[the remark this answers is not in the document]
Comment (remydenton) [reply packages/components/bolt-share/src/share.twig]: Right, I get why you need some attributes on the outer container.  But do you necessarily need any of them on the inner container?

File: packages/components/bolt-share/src/share.twig (hunk 1 above)
Comment (mikemai2awesome) [reply packages/components/bolt-share/src/share.twig]: But I want to define the text to be `Copy share link` though, and have it translatable. If it comes from the schema default, it's not translated. Also, in this case, I am wrapping something around the text so I can style some hover animation.

This feels dirty, I need a better way:
```
{% include "@bolt-components-copy-to-clipboard/copy-to-clipboard.twig" with {
  text: "<span class='c-bolt-share__link-text'>#{copy_to_clipboard_text}</span>",
  url: copy_to_clipboard_url,
  attributes: {
    class: "c-bolt-share__link"
  }
} only %}
```

**Comment by remydenton** [comment]
When you deprecate something in a PR, it's important to follow _all_ the steps in part I of the [ deprecation wiki page](https://github.com/bolt-design-system/bolt/wiki/Deprecating-component-properties#i-for-bolt-prs-that-introduce-a-deprecation)

Missing pieces here:
- Update your schema to specifically exclude the old properties.  This lets us quickly find any incorrect usages in pattern lab:
```yml
not:
  required:
    - copyToClipboard
```
- Add a description of the deprecations to you PR description.  This will be used by the Drupal guys when pulling this in.

Yes, deprecation is a pain in the butt.

**Pull request merged**